apis/google/gmail/code/readEmail.py

```diff
@@ -0,0 +1,63 @@
+import os
+
+from googleapiclient.errors import HttpError
+
+from auth import client
+
+
+def main():
+    email_id = os.getenv('EMAIL_ID')
+    email_subject = os.getenv('EMAIL_SUBJECT')
+    if email_id is None and email_subject is None:
+        raise ValueError("Either email_id or email_subject must be set")
+
+    service = client('gmail', 'v1')
+    try:
+        if email_subject is not None:
+            query = f'subject:"{email_subject}"'
+            response = service.users().messages().list(userId='me', q=query).execute()
+            if not response:
+                raise ValueError(f"No emails found with subject: {email_subject}")
+            email_id = response['messages'][0]['id']
+
+        msg = service.users().messages().get(userId='me', id=email_id).execute()
+        body = get_email_body(msg)
+
+        subject = None
+        sender = None
+        for header in msg['payload']['headers']:
+            if header['name'] == 'Subject':
+                subject = header['value']
+            if header['name'] == 'From':
+                sender = header['value']
+
+        print(f'From: {sender}, Subject: {subject}')
+        print(f'Body:\n{body}')
+
+    except HttpError as err:
+        print(err)
+
+
+def get_email_body(message):
+    import base64
+    try:
+        # Check if it's a multipart message
+        if 'parts' in message['payload']:
+            for part in message['payload']['parts']:
+                # Only extract text/plain content (you can also extract 'text/html' if needed) @TODO: add this
```

Comment:
Does this need to be done prior to merging? Not sure if it's a requirement.

Reply:
i dont think so, but i went ahead and did it anyway. not sure how well it will work for large html emails, but its ok for a first pass